docs/assertions/custom-assertions.mdx

```diff
@@ -0,0 +1,239 @@
+---
+title: Custom Assertions
+---
+
+Pester allows users to create their own `Should`-operators for more advanced assertions. This is done by defining a test-function and registering it with Pester using the guidelines below.
+
+## Create the function
+
+### Requirements / Interface
+
+[Should](../commands/Add-ShouldOperator) will automatically pass on values from pipeline to your function as `ActualValue`.
+This can be an array or a single object depending on whether you register the function with `-SupportsArrayInput`.
+
+When `Should` calls your custom assertion, it will invoke it with the following parameters:
+- **ActualValue**: The value passed to `Should` to assert. 
+- **Negate**: Internal switch/bool parameter used to delcare if the user called `Should -Not ...`.
+- **CallerSessionState**: Used by some internal assertions to invoke code in the user's state. This can usually be ignored.
+- *Remaining bound parameters*, typically matching the parmaeters your accept in your assertion, like `ExpectedValue`, `Because`, `MyCustomParameter` etc.
+
+The function should return a object with the following properties:
+- **Succeeded**: Bool result of assert
+- **FailureMessage**: Message to the user explaining why the assert failed, typically in the format `Expected <expected value>, but got <actual value>`. If the function supports the commonly used `-Because` parameter, this property should include that message.
+
+If a problem occurs in your function, ex. it was called with invalid parameters, throw an exception. This will also fail the `Should` assertion, while making it clear to the user that it's not normal behavior.
+
+:::tip ActualValue is provided as a single parameter
+`Should` accepts the value to assert by pipeline but passes it to the assertion using `-ActualValue $ActualValue`. This means `ValueFromPipelineByPropertyName` is not available inside your function, but you can write your own logic to work with complex types provided to `ActualValue`.
+::: 
+
+:::caution Using advanced functions as assertions
+Advanced functions, typically enabled by using `[CmdletBinding()]` or `[Parameter()]`, disallows arguments by default. In this scnario, make sure to accept all mandatory parameters in your param-block to avoid errors even if you don't use them. They are `ActualValue`, `Negate` and `CallerSessionState`.
```

Comment:
```suggestion
Advanced functions, typically enabled by using `[CmdletBinding()]` or `[Parameter()]`, disallows arguments by default. In this scenario, make sure to accept all mandatory parameters in your param-block to avoid errors even if you don't use them. They are `ActualValue`, `Negate` and `CallerSessionState`.
```